### ISSUE TITLE
Implement Selection Shortcuts

### ISSUE THREAD

**Issue by chinakov** [audacity/audacity#8044]
### Problem to be solved

Let's implement shortcuts for selection from Audacity 3. 

### Your idea

We should duplicate the functionality that is present in Audacity 3. 

Selection Start - Left Arrow
Selection End - Right Arrow
Seleciton Start - Ctrl + [
Selection End - Ctrl + ]
Selection to Start - Shift+Home
Selection to End - Shift+End
Selection Extend Left - Shift + Left
Selection Extend Right - Shift + Right
Selection Contract Left - Ctrl+Shift+Right
Selection Contract Right - Ctrl+Shift+Left

Latest Figma design: https://www.figma.com/design/WllUQSlNZ7pQpsFLfo5ldv/Menus?node-id=202-1503&m=dev

**Comment by ianalarina** [comment]
@Tantacrul @LWinterberg @DilsonsPickles - please, sign off the shortcuts list. I remember there were discussions about changing them/implementing new shortcuts

**Comment by DilsonsPickles** [comment]
No issues from me, I can't see why we would want to change selection shortcuts. @LWinterberg any objections?

**Comment by DavidBailes** [comment]
> We should duplicate the functionality that is present in Audacity 3.

@DilsonsPickles . I think the set of commands that are useful for selecting a time range depends on the relationship between the cursor and any selected time range.

In AU3 they are not independent - there is either a cursor position (selection start and end are the same) or a time range (selection end is greater than selection start). For example, if there is a time range, you can't also set the "cursor" to some arbitrary position, for example, some time after the end of the time range.

In Reaper, the edit cursor and the time selection are independent. If you have a time selection, you can move the edit cursor to wherever you want, and this has no effect on the time selection.

In AU4, I can't work out what the relationship is meant to be. If there's a time selection I can move the cursor either by dragging it or using left/right arrows without affecting the time selection. But if I press the rewind to start button, the cursor moves to time zero, but the time selection is removed - they are not independent. Could you explain what the user model for the relationship between the cursor and any time selection is meant to be?

**Comment by LWinterberg** [comment]
We designed the playhead/selection relationship such that the playhead is more free to be placed and moved around, while also broadly emulating AU3's behaviors. It's perhaps not ideal, but so far it's also not confused people during user testing.

**Comment by DavidBailes** [comment]
> We designed the playhead/selection relationship such that the playhead is more free to be placed and moved around, while also broadly emulating AU3's behaviors. It's perhaps not ideal, but so far it's also not confused people during user testing.

Is there any specification for the ways in which "the playhead is more free to be placed and moved around" which differs from AU3?

**Comment by LWinterberg** [comment]
The way it currently works is intended, we haven't written a documentation yet. It seems to me though that you figured out the behavior already?

**Comment by DavidBailes** [comment]
> The way it currently works is intended, we haven't written a documentation yet. It seems to me though that you figured out the behavior already?

Using the keyboard, the only difference I've figured is that currently left/right arrows can move the playhead independently of the time selection. So I'm assuming that this is the only difference for keyboard users. However there are a couple of issues with this:

1. If I use the left/right arrow keys to move the playhead away from the start of the time selection, then I don't understand the significance of the playhead position when this is the case. For example, if you start playback, playback does not start from the position of the playhead - just the selected time range is played. If I use the spit command (Ctrl+I), then the time selection is used for the split, not the playhead position. (The same issue arises if you dragged the playhead using the mouse)

2. In the list of shortcuts to be implemented are:
Selection Start - Left Arrow
Selection End - Right Arrow
rather than moving the playhead.

**Comment by DilsonsPickles** [comment]
Apologies, I had a lot on 2 weeks ago and didn't give this the full attention it deserved. 

@DavidBailes you are correct. 

Selection start/end using the left/right arrow now doesn't work because our new playhead is persistent. 

For context, one of my reasons for implementing a persistent playhead was to help improve the experience of scrubbing. Whilst this hasn't been realised into a design specification yet, the broad strokes have been figured out so that we can remove the scrub ruler.

For the selection shortcuts I believe we should start with:

Move playhead cursor left - Left Arrow
Move playhead cursor right - Right Arrow
Create selection to the left of playhead cursor - Shift + Left arrow (with no present selection)
Create selection to the right of playhead cursor - Shift + Left arrow (with no present selection)
Selection Start AND move playhead cursor to start of selection AND bring it into viewport  - Ctrl + [
Selection End AND move playhead cursor to end of selection AND bring it into viewport - Ctrl + ]
Selection to Start (create selection from playhead cursor to start of project) - Shift+Home
Selection to End (create selection from playhead cursor to end of project) - Shift+End
Selection Extend Left - Shift + Left (with existing range selection)
Selection Extend Right - Shift + Right (with existing range selection)
Selection Contract Left - Ctrl+Shift+Right
Selection Contract Right - Ctrl+Shift+Left

Let's get these implemented as a first draft and then have a play with it.

**Comment by adammw** [comment]
Not on that list but I considered surprising when it didn't work - plain Home/End keys to move the cursor.

**Comment by luapmartin** [comment]
maybe we should handle the menu entries too

<img width="393" height="321" alt="Image" src="https://github.com/user-attachments/assets/59df0a9a-63a8-497b-b5c7-8026b5115772" />

in https://github.com/audacity/audacity/pull/11474 the jump to selection start/end action have been added already


note that there are also view focus change shortcut to adjust the view to either the start/end of a selection 

<img width="418" height="237" alt="Image" src="https://github.com/user-attachments/assets/2fee9eee-3d08-4750-bb50-8b569df53973" />
